### PR TITLE
Feature/api key support

### PR DESCRIPTION
**Pull request by solomon-negusse**
The API Key support is added in https://github.com/resource-watch/aqueduct-analysis-microservice/pull/94/commits/40ecdd533379c7ae13ff8e8f3661e9315b85c3bd#diff-38e07dc6ef204210d8a141ad4b679689e0f3ba1058456199b7448559831e0b4e

Python and dependency upgrade as well as linting https://github.com/resource-watch/aqueduct-analysis-microservice/pull/94/commits/9a78f0fb617ab773fa5248369dbbeb1434e28d4e?diff=split&w=1

### CODE REVIEW

**Comment by solomon-negusse** [comment]
Hi @carlosdelrealwri, we are adding API key support to all microservices in the Resource Watch infrastructure.  This PR adds that and upgrades dependencies mainly to enable this.

File: requirements_dev.txt [hunk 1]
```diff
@@ -1,5 +1,7 @@
-pytest==5.2.1
+pytest
```
Comment (tiagojsag) [inline requirements_dev.txt]: pin a version

File: requirements.txt [hunk 2]
```diff
@@ -1,23 +1,22 @@
...
+geopandas==0.13.2
 rtree==0.9.7
 simplejson==3.16.0
-Flask==1.1.1
-RWAPIMicroservicePython==1.0.0
+Flask==2.3.2
+RWAPIMicroservicePython==3.0.0
 requests==2.22.0
 geojson==1.3.5
 pytz
```
Comment (tiagojsag) [inline requirements.txt]: while you're at it, pin all of these too

**Review by tiagojsag — CHANGES_REQUESTED** [review]
There's also a docker-compose-gr.yml file that I dont know what's for, but needs to be either updated with the new env vars, or deleted altogether.

**Pull request merged**